beacon_node/network/src/sync/network_context.rs

```diff
@@ -296,12 +296,20 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         }
     }
 
+    /// Request block of `block_root` if necessary by checking:
+    /// - If the da_checker has a pending block from gossip or a previous request
+    ///
+    /// Returns false if no request was made, because the block is already imported
     pub fn block_lookup_request(
         &mut self,
         lookup_id: SingleLookupId,
         peer_id: PeerId,
-        request: BlocksByRootSingleRequest,
+        block_root: Hash256,
     ) -> Result<bool, &'static str> {
+        if self.chain.data_availability_checker.has_block(&block_root) {
```

Comment:
don't we want to use the `reqresp_pre_import_cache`? Otherwise we'll miss blocks that we're in the process of verifying execution

Reply:
the check in `stable` is essentially this because `data_availability_checker.has_block()` used to check the processing cache

Reply:
lion agreed offline, so I've pushed the change here: https://github.com/sigp/lighthouse/pull/5681/commits/5db3e1b5e5ed91886cad9931b6905282c40fe858